WordPress/Classes/ViewRelated/Site Creation/Wizard/SiteCreationWizardStepOrderer.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+import AutomatticTracks
+
+struct SiteCreationWizardStepOrderer {
+    let stepInvoker: SiteCreationWizardStepInvoker
+    let siteIntentVariant: SiteIntentAB.Variant
+    let siteNameVariant: Variation
+
+    private var shouldShowSiteIntent: Bool {
+        return siteIntentVariant == .treatment && FeatureFlag.siteIntentQuestion.enabled
+    }
+
+    private var shouldShowSiteName: Bool {
+        return siteNameVariant == .treatment(nil) && FeatureFlag.siteName.enabled
+    }
+
+    lazy var steps: [WizardStep] = {
+        guard shouldShowSiteIntent else {
+            return [
+                stepInvoker.designStep,
+                stepInvoker.addressStep,
+                stepInvoker.siteAssemblyStep
+            ]
+        }
+
+        guard shouldShowSiteName else {
+            return [
+                stepInvoker.intentStep,
+                stepInvoker.designStep,
+                stepInvoker.addressStep,
+                stepInvoker.siteAssemblyStep
+            ]
+        }
+
+        return [
+            stepInvoker.intentStep,
+            stepInvoker.nameStep,
+            stepInvoker.designStep,
+            stepInvoker.siteAssemblyStep
+        ]
```

Comment:
Understandable that, given the nested conditions to show site intent and site name, we have this very specialized implementation here, maybe we can just put a reminder that, once we have ruled out what the final flow will be, this will be simplified?

Reply:
That sounds good. I hope it will also make it simpler for future experiments that may shift order.